src/code-snippet/code-snippet.component.ts

```diff
@@ -137,11 +138,19 @@ export class CodeSnippet {
 	@ViewChild("code") code;
 
 	get shouldShowExpandButton() {
-		return this.code ? this.code.nativeElement.getBoundingClientRect().height > 255 : false;
+		// Checks if `hasExpand` button has been initialized in `AfterViewInit` before detecting whether or not to
+		// show the expand button when the code displayed in the component changes during the life of the component.
+		// This is to avoid the `ExpressionChangedAfterItHasBeenCheckedError`.
+		if (this.hasExpandButton === null) {
+			return this.hasExpandButton;
+		}
+		return this.code ? this.code.nativeElement.getBoundingClientRect().height > 255 : false && this.display === "multi";
```

Comment:
Could we wrap this up in a protected method? That way we aren't duplicating code all over the place

Reply:
Just the `this.code ? this.code.nativeElement.getBoundingClientRect().height > 255 : false && this.display === "multi"` part? Any ideas for the name of this function ? 😅

Reply:
Actually ... couldn't we just use this (`shouldShowExpandButton`) in `ngAfterViewInit`?

Reply:
Good idea!!

Reply:
Actually this only works after `hasExpandButton` has been initialized otherwise it will just keep returning null

Reply:
Right ... maybe `canExpand` or something for the method name?